reports/calculator.py

```diff
@@ -1,6 +1,12 @@
+import hashlib
 from marketplace import marketplace
 
 
+def get_raw_data_hash(investment_raw_data):
+    string_unique_dates = ''.join([ str(date) for date in investment_raw_data[marketplace.FILE_DATE].unique()])
```

Comment:
Some thoughts here:
- join accepts an iterable, I don't see the need to explicitly construct the list
- converting str to date, will it work always the same way? What format is using to express the date as a string? I'm thinking if this could be locale dependent.
- Depends on the order that unique is return, the hash could be different right?
I would recommend two things, first to separate the function in two:
- one function that as an input has investment_raw_data and as an output string_unique_dates
- one function that as an input has string_unique_dates and as an output the hash.
Then adding some tests for the first function will make it more clear how this is behaving.

Reply:
I think that I revisited all your points.